File: app/src/main/java/com/example/wego_travel/Models/Pengguna.java
package com.example.wego_travel.Models;

import java.util.Date;

public class Pengguna {
    private int id_pengguna;
    private String name;
    private String email;
    private String password;
    private Jenis_Kelamin jenisKelamin;
    private Date tanggalLahir;
    private String nomorTelepon;
    private String alamat;

    public Pengguna(int id_pengguna, String name, String email, String password, Jenis_Kelamin jenisKelamin, Date tanggalLahir, String nomorTelepon, String alamat) {
        this.id_pengguna = id_pengguna;
        this.name = name;
        this.email = email;
        this.password = password;
        this.jenisKelamin = jenisKelamin;
        this.tanggalLahir = tanggalLahir;
        this.nomorTelepon = nomorTelepon;
        this.alamat = alamat;
    }

    // id pengguna
    public int getIdPengguna(){
        return id_pengguna;
    }

    public void setId_pengguna(int id_pengguna){
        this.id_pengguna = id_pengguna;
    }

    // name
    public String getName(){
        return name;
    }

    public void setName(String name){
        this.name = name;
    }

    // email
    public String getEmail(){
        return email;
    }

    public void setEmail(String email){
        this.email = email;
    }

    // password
    public String getPassword(){
        return password;
    }

    public void setPassword(String password){
        this.password = password;
    }

    // jenis kelamin
    public Jenis_Kelamin getJenisKelamin(){
        return jenisKelamin;
    }

    public void setJenisKelamin(Jenis_Kelamin jenisKelamin) {
        this.jenisKelamin = jenisKelamin;
    }

    // tanggal lahir
    public Date getTanggalLahir(){
        return tanggalLahir;
    }

    public void setTanggalLahir(Date tanggalLahir){
        this.tanggalLahir = tanggalLahir;
    }

    // nomor telepon s
    public String getNomorTelepon(){
        return nomorTelepon;
    }

    public void setNomorTelepon(String nomorTelepon){
        this.nomorTelepon = nomorTelepon;
    }

    // alamat s
    public String setAlamat(){
        return alamat;
    }

    public void setAlamat(String alamat){
        this.alamat = alamat;
    }
}

File: app/src/main/java/com/example/wego_travel/Models/Pengguna.kt
package com.example.wego_travel.Models

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.util.Date

@Parcelize
data class Pengguna (
    @SerializedName("id_pengguna")
    var id_pengguna: Int,
    @SerializedName("nama")
    var nama:String,
    @SerializedName("email")
    var email: String,
    @SerializedName("password")
    var password:String,
    @SerializedName("Jenis_kelamin")
    var Jenis_kelamin:String,
    @SerializedName("Tanggal_lahir")
    var Tanggal_lahir:Date?,
    @SerializedName("nomor_telepon")
    var nomor_telepon:String,
    @SerializedName("alamat")
    var alamat:String
) : Parcelable {
    companion object {
        @Volatile
        private var instance: Pengguna? = null

        fun getInstance(): Pengguna = instance ?: synchronized(this) {
                instance ?: Pengguna(
                    0, "", "", "", "", Date(), "", ""
                )
        }

        fun setInstance(pengguna: Pengguna) {
            instance = pengguna
        }
    }
}